src/test/java/org/carlspring/cloud/storage/s3fs/path/ToUriTest.java

```diff
@@ -58,6 +58,26 @@ void toUri()
         assertEquals(path, pathActual);
     }
 
+    @Test
+    public void toUriSpecialChars()
+    {
+        Path path = getPath("/bucket/([fol! @#$%der])");
```

Comment:
From what I understand, this test should actually be called `toUriWithSpecialCharactersThatShouldNodBeDecoded` instead, right?

Could you please add a test case where there are actual characters that need to be decoded?